packages/medusa/src/api/routes/admin/stock-locations/utils/join-sales-channels.ts

```diff
@@ -14,9 +14,14 @@ const joinSalesChannels = async (
       const salesChannelIds = await channelLocationService.listSalesChannelIds(
         location.id
       )
-      const [salesChannels] = await salesChannelService.listAndCount({
-        id: salesChannelIds,
-      })
+      const [salesChannels] = await salesChannelService.listAndCount(
+        {
+          id: salesChannelIds,
+        },
+        {
+          take: salesChannelIds.length,
```

Comment:
todo: I would suggest to pass just an empty object for the second argument just for the default config value to be avoided. wdyt?